src/datachain/lib/utils.py

```diff
@@ -28,3 +30,19 @@ def __init__(self, message):
 class DataChainColumnError(DataChainParamsError):
     def __init__(self, col_name, msg):
         super().__init__(f"Error for column {col_name}: {msg}")
+
+
+def normalize_col_names(col_names: Sequence[str]) -> dict[str, str]:
+    default_column = 0
+    converted_col_names = {}
+    for column in col_names:
+        original_column = column
+        column = column.lower()
+        column = re.sub("-+", "_", column)
```

Comment:
C: I've changed the rule a bit here as well. It now changes `-` into `_` and multiple `_` into a single one. 

Also checks if the column name after normalization exists it now also applies the `default_column` generator.

Reply:
@shcheklein, quick question: what happens if the original column name _begins_ with a number, would it break something?

Reply:
Asking just in case we use these identifiers as e.g. Python attributes, where `column1` would be valid but `1column` wouldn't

Reply:
good question, I'm a bit hesitant to go that far (not always we have to create a pydantic model - e.g. we might not need for top level fields - just a flat parquet / CSV). I feel it's a bit broader topic ... I'll think about it and if we need to solve it in this PR

Reply:
probably at least in the `dict_to_data_model` we have to ensure this I guess, so can introduce an additional param for that

Reply:
Not sure if relevant, but in the same vein as column names beginning with a number aren't valid, reserved words like `return` shouldn't be valied either. 🙈